### PR TITLE
update API and examples for egpio

### PR DESCRIPTION
**Pull request by ChrisDick**
also fix units of Battery current.

# Description

add new elements to the API for the the LS gpio and update examples. correct the units for the battery charge current from mA to A.

### CODE REVIEW

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 1]
```diff
@@ -218,7 +218,7 @@ You can see a more comprehensive example in [`dialog.py`](https://github.com/emf
 
 Each hexpansion has:
 
-- 5 low speed (LS) emulated GPIO (eGPIO) pins which you can use with [the tildagonos `Pin`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagonos.py) (2, 3, 7, 8, 9)
+- 5 low speed (LS) external GPIO (egpio) pins which you can use with the [`tildagon.Pin`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagonos.py)
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:221]:
```suggestion
- 5 low speed (LS) external GPIO (eGPIO) pins which you can use with the [`tildagon.Pin`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagonos.py)
```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 2]
```diff
@@ -337,16 +337,17 @@ A more elaborate example is this [breadboard tester app](https://github.com/npen
 
 ### Methods
 
-GPIO pins support the standard [`machine.Pin` methods](https://docs.micropython.org/en/latest/library/machine.Pin.html).
+gpio pins support the standard [`machine.Pin` methods](https://docs.micropython.org/en/latest/library/machine.Pin.html). These are refered to as high speed pins as they are connected directly to the ESP32.
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:340]:
```suggestion
GPIO pins support the standard [`machine.Pin` methods](https://docs.micropython.org/en/latest/library/machine.Pin.html). These are refered to as high speed pins as they are connected directly to the ESP32.
```

File: docs/tildagon-apps/reference/badge-hardware.md [hunk 3]
```diff
@@ -337,16 +337,17 @@ A more elaborate example is this [breadboard tester app](https://github.com/npen
...
 
-[eGPIO pins](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagon/pins.py) support the following methods:
+[egpio pins](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagon/pins.py) are refered to as low speed pins as they are connected to a port expander by I2C and support the following methods:
```
Comment (npentrel) [inline docs/tildagon-apps/reference/badge-hardware.md:342]:
```suggestion
[eGPIO pins](https://github.com/emfcamp/badge-2024-software/blob/main/modules/tildagon/pins.py) are refered to as low speed pins as they are connected to a port expander by I2C and support the following methods:
```

**Review by npentrel — APPROVED** [review]
Lgtm % nits. Is there a reason you'd prefer to change the capitalisation of gpio? I would like to keep it consistent and currently it's capitalised everywhere

**Comment by ChrisDick** [comment]
I renamed it to match the name of the module. Do you think that matters? not sure it's wise changing the name in the sw, but we could while it's still in rc. I don't mind either way, happy to replace all occurrences in the docs if that makes more sense to keep consistency. Another solution is to make them all eGPIO except in the methods section and make that one egpio.ePin?

**Comment by npentrel** [comment]
@ChrisDick I think let's use `egpio` and `gpio` in code formatting when referring to the class and eGPIO and GPIO when referring to the concept

**Comment by npentrel** [comment]
Merging for now

**Pull request merged**